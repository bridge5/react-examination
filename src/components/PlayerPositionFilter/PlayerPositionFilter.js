import React from 'react';
import './PlayerPositionFilter.css';
const positions = ['ALL', 'SF', 'PG'];
const PlayerPositionFilter = ({ setPosition }) => {
  return (
    <ol className="breadcrumb">
      {positions.map((position, index) => (
        <li key={index} className="breadcrumb-item">
          <button
            type="button"
            className="btn-link"
            onClick={() => setPosition(position)}
          >
            {position}
          </button>
        </li>
      ))}
    </ol>
  );
};

export default PlayerPositionFilter;
