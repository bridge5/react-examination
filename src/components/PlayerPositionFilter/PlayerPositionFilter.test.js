import React from 'react';
import { shallow } from 'enzyme';
import PlayerPositionFilter from './PlayerPositionFilter';

describe('PlayerPositionFilter feature', () => {
  it('render PlayerPositionFilter correctly', () => {
    const wrapper = shallow(<PlayerPositionFilter />);
    expect(wrapper).toMatchSnapshot();
  });
});
