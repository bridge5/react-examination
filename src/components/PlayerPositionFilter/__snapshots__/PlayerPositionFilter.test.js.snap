// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerPositionFilter feature render PlayerPositionFilter correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlayerPositionFilter />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="breadcrumb-item"
        >
          <button
            className="btn-link"
            onClick={[Function]}
            type="button"
          >
            ALL
          </button>
        </li>,
        <li
          className="breadcrumb-item"
        >
          <button
            className="btn-link"
            onClick={[Function]}
            type="button"
          >
            SF
          </button>
        </li>,
        <li
          className="breadcrumb-item"
        >
          <button
            className="btn-link"
            onClick={[Function]}
            type="button"
          >
            PG
          </button>
        </li>,
      ],
      "className": "breadcrumb",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="btn-link"
            onClick={[Function]}
            type="button"
          >
            ALL
          </button>,
          "className": "breadcrumb-item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ALL",
            "className": "btn-link",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "ALL",
          "type": "button",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="btn-link"
            onClick={[Function]}
            type="button"
          >
            SF
          </button>,
          "className": "breadcrumb-item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SF",
            "className": "btn-link",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "SF",
          "type": "button",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="btn-link"
            onClick={[Function]}
            type="button"
          >
            PG
          </button>,
          "className": "breadcrumb-item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "PG",
            "className": "btn-link",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "PG",
          "type": "button",
        },
        "type": "li",
      },
    ],
    "type": "ol",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="breadcrumb-item"
          >
            <button
              className="btn-link"
              onClick={[Function]}
              type="button"
            >
              ALL
            </button>
          </li>,
          <li
            className="breadcrumb-item"
          >
            <button
              className="btn-link"
              onClick={[Function]}
              type="button"
            >
              SF
            </button>
          </li>,
          <li
            className="breadcrumb-item"
          >
            <button
              className="btn-link"
              onClick={[Function]}
              type="button"
            >
              PG
            </button>
          </li>,
        ],
        "className": "breadcrumb",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn-link"
              onClick={[Function]}
              type="button"
            >
              ALL
            </button>,
            "className": "breadcrumb-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ALL",
              "className": "btn-link",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "ALL",
            "type": "button",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn-link"
              onClick={[Function]}
              type="button"
            >
              SF
            </button>,
            "className": "breadcrumb-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SF",
              "className": "btn-link",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "SF",
            "type": "button",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn-link"
              onClick={[Function]}
              type="button"
            >
              PG
            </button>,
            "className": "breadcrumb-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "PG",
              "className": "btn-link",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "PG",
            "type": "button",
          },
          "type": "li",
        },
      ],
      "type": "ol",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
